hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/BlockRetentionManager.java

```diff
@@ -0,0 +1,170 @@
+     */
+    public void startCleanup() {
+        CompletableFuture.runAsync(this::cleanupExpiredBlocks, cleanupExecutor).exceptionally(ex -> {
+            log.warn("Error during cleanup: {}", ex.getMessage());
+            return null;
+        });
+    }
+
+    /**
+     * Deletes all block files that have expired.
+     */
+    public void cleanupExpiredBlocks() {
+        // Collect files into a list to avoid consuming the stream multiple times
+        final List<Path> fileList = listFiles()
+                .filter(file -> isBlockFile(file) && isFileExpired(file))
+                .toList();
+
+        // Submit deletion tasks for each file
+        final List<CompletableFuture<Void>> futures = fileList.stream()
+                .map(file -> CompletableFuture.runAsync(() -> deleteFile(file), cleanupExecutor))
+                .toList();
+
+        // Wait for all deletion tasks to complete
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+    }
+
+    /**
+     * Shuts down the underlying scheduler and cleanup executor services.
+     */
+    public void shutdown() {
+        scheduler.shutdown();
+        cleanupExecutor.shutdown();
+        try {
+            if (!scheduler.awaitTermination(60, TimeUnit.SECONDS)) {
+                scheduler.shutdownNow();
+            }
+            if (!cleanupExecutor.awaitTermination(60, TimeUnit.SECONDS)) {
+                cleanupExecutor.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            scheduler.shutdownNow();
+            cleanupExecutor.shutdownNow();
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private Stream<Path> listFiles() {
+        Stream<Path> files = Stream.empty();
+        try {
+            files = Files.list(uploadedDir);
+        } catch (IOException ex) {
+            log.warn("Error scanning directory: {}", ex.getMessage());
+        }
+
+        return files;
+    }
+
+    private void deleteFile(@NonNull final Path file) {
+        try {
+            Files.delete(file);
+        } catch (IOException ex) {
+            log.warn("Failed to delete file {}: {}", file, ex.getMessage());
+        }
+    }
+
+    private boolean isBlockFile(@NonNull final Path file) {
+        final String fileName = file.getFileName().toString();
+        return fileName.endsWith(BLOCK_FILE_EXTENSION) || fileName.endsWith(BLOCK_FILE_EXTENSION_GZ);
+    }
+
+    private boolean isFileExpired(@NonNull final Path file) {
+        Instant fileTime;
+        try {
+            fileTime = Files.getLastModifiedTime(file).toInstant();
+        } catch (IOException e) {
+            log.warn("Failed to get last modified time for file {}: {}", file, e.getMessage());
+            fileTime = Instant.MAX;
+        }
+
+        final Instant now = Instant.now();
+        final Instant expirationTime = now.minus(retentionPeriod);
+        return fileTime.isBefore(expirationTime);
```

Comment:
This will always return `false` if `fileTime` is `Instant.MAX`, yes? Why not return `false` on line 163 instead of setting `fileTime = Instant.MAX`?